### PR TITLE
Checkout any submodules when running Actions

### PR DESCRIPTION
**Pull request by PeterAdams-A**
Motivation:

Some repos using this as their source of GitHub Actions logic have submodules.

Modifications:

Change workflows to checkout submodules.

Result:

All CI will work cleanly

### CODE REVIEW

File: .github/workflows/swift_matrix.yml [hunk 1]
```diff
@@ -168,6 +169,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           persist-credentials: false
+          format_check_enabled: true
```
Comment (gjcairo) [inline .github/workflows/swift_matrix.yml:172]: Nit but do we want to do this in a separate PR, since it's not related to checking out submodules?

Comment (PeterAdams-A) [reply .github/workflows/swift_matrix.yml:172]: Yikes, didn't mean to do that.  Will take ti out.

**Review by ali-ahsan-ali — APPROVED** [review]
